Add graphing and filtering to student report

jhilde opened the pull request:
This is the consolidated pull request for phase 2 (graphs) and 3 (filtering) of the new student report.

Supersedes pull request #873 
[student_report.pdf](https://github.com/studentinsights/studentinsights/files/856881/student_report.pdf)

jhilde commented:
Pinging on this @alexsoble @kevinrobinson. Let me know if you have feedback.

alexsoble commented on app/controllers/students_controller.rb:
```diff
@@ -144,29 +148,52 @@ def student_feed(student, restricted_notes: false)
+    @student = Student.find(params[:id])
+    @current_educator = current_educator
+    @url = root_url.chomp('/') + request.path
+    @sections = (params[:sections] || "").split(/\s*,\s*/)
```
@jhilde Question -- what is the `.split(/\s*,\s*/)` code doing?

jhilde replied:
Should be splitting on comma separated values. The sections param is a comma delimited string of section names. It ignores leading and trailing white space which may be irrelevant.

alexsoble replied:
OK cool, yeah I was wondering if `.split(",")` could do the job just as well.

jhilde replied:
Switched to the simpler .split(",")

alexsoble commented on app/controllers/students_controller.rb:
```diff
@@ -144,29 +148,52 @@ def student_feed(student, restricted_notes: false)
+  def set_up_student_report_data
+    @student = Student.find(params[:id])
+    @current_educator = current_educator
+    @url = root_url.chomp('/') + request.path
```
@jhilde Quick question, what is `@url = root_url.chomp('/') + request.path` doing? Thanks!

jhilde replied:
I realize that's a holdover from @erose's initial take on the report. Initially the url was included in the footer of the report. I removed it to save space.

So we can remove the variable as well.

alexsoble replied:
@jhilde Great, yeah let's take it out if it's not used.

jhilde replied:
Removed superfluous code.

alexsoble commented:
Hi @jhilde!  

I left a couple of in-line code comments about specific lines of code I want to understand a little better.

Also posting a GIF of what this new filtering UI looks like, below. I recommend posting screenshots and/or GIFs along with your PRs, it makes it clearer what the code you're merging in does.  

# GIF
![new-student-report-ui](https://cloud.githubusercontent.com/assets/3209501/24324878/a0167d9a-1164-11e7-8d54-6be3844a4183.gif)

What are your thoughts on the UI Justin? 

# Code duplication question

I'm noticing some of of the functions on `student_profile_pdf.js` are the same as functions on `profile_bar_chart.js`: `monthKeys`, `defaultMonthKey`, `eventsToMonthBuckets`. What do you think about extracting these methods out so that both of the files can share the same methods?

Cheers -- hope these questions and feedback are useful. Let me know what other ideas, thoughts, or questions are in your brain!

alexsoble commented:
P.S.  @jhilde let me know if you want to jump on a screenshare or hangout some time to talk about rebasing! We can use rebasing to get rid of some of the "noisy" commits that say things like `Merge branch 'master' of https://github.com/jhilde/studentinsights` and leave us an easier-to-read history.

jhilde commented:
@alexsoble Thanks for the helpful feedback! Those functions are exactly the same for that purpose. I'd be happy to factor them out. Thoughts about where they should belong?

And yes, I'd love to work through rebase and norms around verbosity on check in comments, etc.

jhilde commented:
And @alexsoble, what did you use to create the animated gif. Very cool.

alexsoble commented:
@jhilde LICEcap! http://www.cockos.com/licecap/

alexsoble commented:
@jhilde Let's figure out a good time to jump on a screenshare over email!

kevinrobinson commented:
Hi @jhilde, sorry for the delay on reviewing, too much traveling the last two weeks.  And thanks for jumping in @alexsoble so we can keep moving things forward.  🌮 👍

jhilde commented:
Moved shared graphing functions into a helper app/assets/javascripts/helpers/graph_helpers.js, removed code from app/assets/javascripts/student_profile/profile_bar_chart.js and pdf/student_profile_pdf.js, reworked the tests.

alexsoble commented:
Nice job with the rebase @jhilde! Will merge in and push to prod early next week to avoid any Friday late afternoon push-to-prod jinxes!

jhilde commented:
Awesome! Thanks.

alexsoble commented:
🚢 ing

kevinrobinson commented:
💥 awesome @jhilde! 👍
